### PR TITLE
new feature: allow user to setup replication's underlying receive buffer

### PR DESCRIPTION
**Pull request by funkygao**

### CODE REVIEW

File: replication/binlogsyncer.go [hunk 1]
```diff
@@ -57,6 +58,9 @@ type BinlogSyncerConfig struct {
 	ParseTime bool
 
 	LogLevel string
+
+	// RecvBuffer sets the size of the operating system's receive buffer associated with the connection.
+	RecvBuffer int
```
Comment (siddontang) [inline replication/binlogsyncer.go]: I prefer RecvBufferSize

**Comment by siddontang** [comment]
Rest LGTM

File: replication/binlogsyncer.go (hunk 1 above)
Comment (funkygao) [inline replication/binlogsyncer.go]: inside golang net package, SetReadBuffer() is used, instead of SetReadBufferSize

Comment (siddontang) [inline replication/binlogsyncer.go]: The SetReadBuffer argument is `bytes`, so RecvBufferSize makes more sense as the function argument.

**Comment by siddontang** [comment]
Thanks @funkygao

**Pull request merged**